prov/verbs/src/fi_verbs.h

```diff
@@ -597,14 +597,14 @@ static inline int
 fi_ibv_process_wc(struct fi_ibv_cq *cq, struct ibv_wc *wc)
 {
 	/* Handle WR entry when user doesn't request the completion */
-	if (wc->wr_id == VERBS_INJECT_FLAG)
-		return 0;
+	if (wc->wr_id == VERBS_INJECT_FLAG) {
+		int ret;
 
-	if (OFI_UNLIKELY(wc->status == IBV_WC_WR_FLUSH_ERR)) {
-		/* Handle case when remote side destroys
-		 * the connection, but local side isn't aware
-		 * about that yet */
-		return 0;
+		while ((ret = ibv_poll_cq(cq->cq, 1, wc)) > 0) {
+			if (wc->wr_id != VERBS_INJECT_FLAG)
+				return 1;
+		}
```

Comment:
Perhaps we should poll for additional completions outside this function as doing inside may cause completion ordering issues especially since `fi_ibv_poll_reap_unsig_cq` calls `fi_ibv_process_wc` in a loop.

Reply:
Good catch! Thanks. I fixed it by implementing two versions of `fi_ibv_process_wc` function with additional polling (`fi_ibv_process_wc_poll_new`) and without additional polling (`fi_ibv_process_wc`)